### DISCUSSION TITLE
Button to create a component / Transitioning between templates

### DISCUSSION THREAD

**Discussion by dnl0037** [idom-team/idom#1095] [Question] [answered]
Hi, I am new to react, html and css, but have some experience on python. Now, with react.py, I am trying to code a web app and encountered these doubts: 1) is it possible to have a buttom with a function that generates a new component? For example, a buttom that creates a table. 2) How can I transition from one template to another?

**Comment by Archmonger** [comment, marked as answer]
Yes. Both of these are achieved through conditional rendering.

The in-progess docs has [an example](https://reactive-python.github.io/reactpy/learn/quick-start/#conditional-rendering) on this.
